packages/common/src/utils/nft.utils.ts

```diff
@@ -173,4 +185,103 @@ export class NftUtils {
       throw new Error(`onNewNftEvent lambda invoke failed for tx: ${txId}`);
     }
   }
+
+  /**
+   * Process an event from the full node and invoke the NFT handler lambda if needed.
+   * @param eventData The full node event data
+   * @param stage The deployment stage
+   * @param network The network
+   * @param logger The logger instance
+   * @returns A promise that resolves when the processing is complete
+   */
+  static async processNftEvent(
+    eventData: {
+      hash: string;
+      version: number;
+      tokens: string[];
+      token_name?: string | null;
+      token_symbol?: string | null;
+      inputs: Array<{
+        tx_id: string;
+        index: number;
+        spent_output: {
+          value: number;
+          token_data: number;
+          script: string;
+          decoded?: {
+            type: string;
+            address: string;
+            timelock: unknown;
+          } | null;
+        }
+      }>;
+      outputs: Array<{
+        value: number;
+        token_data: number;
+        script: string;
+        decoded?: {
+          type: string;
+          address: string;
+          timelock: unknown;
+        } | null;
+      }>;
+      [key: string]: unknown; // Allow other properties
+    },
+    stage: string,
+    network: unknown,
+    logger: Logger
+  ): Promise<boolean> {
+    // Early return if NFT auto review is disabled
+    if (!isNftAutoReviewEnabled()) {
+      logger.debug('NFT auto review is disabled. Skipping NFT handler invocation.');
+      return false;
+    }
+
```

Comment:
suggestion(non-blocking): we could add a condition here to skip if it's not a tx to create token.

```
if (eventData.version !== constants.CREATE_TOKEN_TX_VERSION) return false;
```